pytorch_lightning/loops/dataloader/evaluation_loop.py

```diff
@@ -366,3 +377,21 @@ def _print_results(results: List[_OUT_DICT], stage: str, file: Optional[IO[str]]
                         lines.append(row_format.format(metric, *row).rstrip())
                 lines.append(bar)
                 print(os.linesep.join(lines), file=file)
+
+
+def _select_data_fetcher_type(trainer: "pl.Trainer") -> Type[AbstractDataFetcher]:
+    lightning_module = trainer.lightning_module
+    step_fx = (
+        getattr(lightning_module, "test_step") if trainer.testing else getattr(lightning_module, "validation_step")
+    )
+    if is_param_in_hook_signature(step_fx, "dataloader_iter", explicit=True):
+        rank_zero_warn(
+            "Found `dataloader_iter` argument in the `training_step`. Note that the support for "
+            "this signature is experimental and the behavior is subject to change."
+        )
+        return DataLoaderIterDataFetcher
+    elif os.getenv("PL_INTER_BATCH_PARALLELISM", "0") == "1":
+        if not isinstance(trainer.accelerator, GPUAccelerator):
+            raise MisconfigurationException("Inter batch parallelism is available only when using NVidia GPUs.")
```

Comment:
nit: let's not copy over this typo

```suggestion
            raise MisconfigurationException("Inter batch parallelism is available only when using Nvidia GPUs.")
```